### PR TITLE
Add JSDoc headers to Auto Generate API documentation

### PR DESCRIPTION
**Pull request by BlackstoneEngineering**
Goal: Add JSDoc style headers to the API calls so that the documentation can be auto-generated.

To generate the source run the build-docs.sh script
The extra .hbs files are necessary to munge the output into a preferred format.

In addition to this PR we will need to add a Github Action to detect changes and push them to the API docs site.

@DEskinazi - Docs team reviewer
@kietdo360 / @imnutz - JS SDK reviewers.

### CODE REVIEW

**Comment by BlackstoneEngineering** [comment]
@imnutz - if necessary we can remove the build script and remove the .hbs files. However, this would prevent the JS SDK team from being able to self generate the API documentation.

File: jsdoc-config.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+                    "plugins/clicks.js",
+                    "plugins/globalid.js",
+                    "plugins/personalization.js",
+                    "plugins/servercookie.js",
+                    "plugins/consent-manager.js"
+                ],
+        // "exclude":["\\test\\"],
+        // "includePattern": ".+\\.js(doc|x)?$",
+        "excludePattern": "(test|out|docs|.github|node_modules|dist)"
+        // "excludePattern": "(^|\\/|\\\\)_"
```
Comment (imnutz) [inline jsdoc-config.json:25]: Can we remove the commented lines?

Comment (BlackstoneEngineering) [reply jsdoc-config.json]: done

File: jsdoc-config.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{
+    "plugins": ["plugins/markdown",
+                "plugins/summarize"
+                // "./node_modules/@ckeditor/jsdoc-plugins/lib/export-fixer/export-fixer.js"
+                ],
+    "recurseDepth": 10,
+    "source": {
+        "include":[
+                    // The order here determines the order displayed in the docs.
```
Comment (imnutz) [inline jsdoc-config.json:9]: Can you remove all of these?

Comment (BlackstoneEngineering) [reply jsdoc-config.json:9]: Line 9 is a valid comment that conveys important information. I suggest leaving it in place.

Comment (imnutz) [reply jsdoc-config.json:9]: This is invalid for .json file, we are not allowed to have comments like this

**Review by tung-vu-td — APPROVED** [review]
LGTM

**Review by imnutz — APPROVED** [review]

**Review by nmpennypacker — APPROVED** [review]
Reviewed for Data Exfiltrations

**Comment by BlackstoneEngineering** [comment]
Deleting branch as part of API Doc Portal Phase 1 cleanup

**Pull request merged**